### PR TITLE
numa fixes

### PR DESCRIPTION
**Pull request by benharsh**
Concerns:
- There's a compiler bug in miniMD where the compiler infers the incorrect type in resolution because an autoCopy of an ArrayView is later inserted (resulting in a non-view). I think this is likely poking some untested and weird behavior in our reduction implementation.
- I do not know what performance testing will look like. I did not implement the indexing-cache for multi-ddata since at this time I'm not sure how it should work.

Basically, these changes should save us from the hundreds/thousands of failures we would otherwise encounter upon merging into chapel-lang/master.

### CODE REVIEW

**Comment by benharsh** [comment]
@bradcray : some numa fixes. Not ideal, but a step in the right direction.

File: modules/internal/ArrayViewReindex.chpl [hunk 1]
```diff
@@ -109,7 +109,7 @@ module ArrayViewReindex {
       }
     }
 
-    iter these(param tag: iterKind) ref where tag == iterKind.standalone {
+    iter these(param tag: iterKind) ref where tag == iterKind.standalone && !localeModelHasSublocales {
```
Comment (bradcray) [inline modules/internal/ArrayViewReindex.chpl]: What happens if the locale model does have sublocales?  It falls back to leader-follower?  Is there precedent for this (in DefaultRectangular, say?)?

File: test/memory/sungeun/refCount/domainMaps.lm-numa.good [hunk 2]
```diff
@@ -77,4 +77,4 @@ Create domain and array:
 	0 bytes leaked
 	0 private objects leaked
 total:
-	1104 bytes leaked
+	0 bytes leaked
```
Comment (bradcray) [inline test/memory/sungeun/refCount/domainMaps.lm-numa.good]: These deltas make me wonder: Can we simply retire all of the special domainMaps*.good files other than the "standard" one now?

File: modules/internal/ArrayViewReindex.chpl (hunk 1 above)
Comment (benharsh) [inline modules/internal/ArrayViewReindex.chpl]: Yes, we should fall back on the leader-follower iterators. I'd rather not have to make this change, but otherwise we run into a variety of issues with reductions over arrays.

I believe there is some precedent for param-choosing iterators in DefaultRectangular based on the locale-model.

File: test/memory/sungeun/refCount/domainMaps.lm-numa.good (hunk 2 above)
Comment (benharsh) [inline test/memory/sungeun/refCount/domainMaps.lm-numa.good]: Hmm, good question. I'd be inclined to merge and let nightly testing notify us if things changed.

Comment (bradcray) [inline test/memory/sungeun/refCount/domainMaps.lm-numa.good]: I'd be fine doing that w.r.t. the question as to whether we can do this for all *.good files.  But it seems like we could retire this one in favor of the generic one now?  (unless I missed something in my visual diff -- it looked like both are now leak-free?)

**Comment by bradcray** [comment]
Merging despite the comment above in order to get performance testing against this version tonight.
